src/client.rs

```diff
@@ -455,28 +485,29 @@ impl Client {
     /// # Example
     ///
     /// ```
-    /// # use meilisearch_sdk::{client::*, errors::Error, key::KeyBuilder};
+    /// # use meilisearch_sdk::{client::*, errors::Error, key::KeyBuilder, key::KeyUpdater};
     /// #
     /// # let MEILISEARCH_HOST = option_env!("MEILISEARCH_HOST").unwrap_or("http://localhost:7700");
     /// # let MEILISEARCH_API_KEY = option_env!("MEILISEARCH_API_KEY").unwrap_or("masterKey");
     /// #
     /// # futures::executor::block_on(async move {
     /// let client = Client::new(MEILISEARCH_HOST, MEILISEARCH_API_KEY);
-    /// let key = KeyBuilder::new("update_key");
-    /// let mut key = client.create_key(key).await.unwrap();
-    /// assert!(key.indexes.is_empty());
+    /// let new_key = KeyBuilder::new();
+    /// let name = "my name".to_string();
+    /// let mut new_key = client.create_key(new_key).await.unwrap();
+    /// let mut key_update = KeyUpdater::new(new_key);
+    /// key_update.with_name(&name);
     ///
-    /// key.indexes = vec!["*".to_string()];
-    /// let key = client.update_key(key).await.unwrap();
-    /// assert_eq!(key.indexes, vec!["*"]);
+    /// let key = client.update_key(key_update).await.unwrap();
+    /// assert_eq!(key.name, Some(name));
     /// # client.delete_key(key).await.unwrap();
     /// # });
     /// ```
-    pub async fn update_key(&self, key: impl AsRef<Key>) -> Result<Key, Error> {
-        request::<&Key, Key>(
+    pub async fn update_key(&self, key: impl AsRef<KeyUpdater>) -> Result<Key, Error> {
+        request::<&KeyUpdater, Key>(
             &format!("{}/keys/{}", self.host, key.as_ref().key),
             &self.api_key,
-            Method::Patch(key.as_ref()),
+            Method::Patch(key.as_ref()), // name and description
```

Comment:
```suggestion
            Method::Patch(key.as_ref()),
```

I don't understand this comment, and it’s stressing me out; maybe we should rephrase it... or maybe we should just remove it?